Gems/AtomTressFX/Assets/Shaders/HairStrands.azsli

```diff
@@ -0,0 +1,181 @@
+/*
+ * Modifications Copyright (c) Contributors to the Open 3D Engine Project. 
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ * 
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT) AND MIT
+ *
+ */
+
+//---------------------------------------------------------------------------------------
+// Shader code related to hair strands in the graphics pipeline.
+//-------------------------------------------------------------------------------------
+//
+// Copyright (c) 2019 Advanced Micro Devices, Inc. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once 
+
+#include <HairUtilities.azsli>
+
+#define CM_TO_METERS_RENDER 0.01
+
+float4 GetSharedVector4(int offset)
+{
+    return float4(
+        float3(
+            asfloat(PassSrg::m_skinnedHairSharedBuffer[offset]),
+            asfloat(PassSrg::m_skinnedHairSharedBuffer[offset + 1]),
+            asfloat(PassSrg::m_skinnedHairSharedBuffer[offset + 2])
+        ),// * CM_TO_METERS,   // convert to meters when using
+        asfloat(PassSrg::m_skinnedHairSharedBuffer[offset + 3])
+    );
+}
+
+float4 GetSharedPosition(int vertexIndex)
+{
+    int  vertexOffset = (HairDynamicDataSrg::m_positionBufferOffset >> 2) + (vertexIndex << 2);
+    return GetSharedVector4(vertexOffset);
+}
+
+float3 GetSharedTangent(int tangentIndex)
+{
+    int tangentOffset = (HairDynamicDataSrg::m_tangentBufferOffset >> 2) + (tangentIndex << 2);
+
+    return float3(
+        asfloat(PassSrg::m_skinnedHairSharedBuffer[tangentOffset]),
+        asfloat(PassSrg::m_skinnedHairSharedBuffer[tangentOffset + 1]),
+        asfloat(PassSrg::m_skinnedHairSharedBuffer[tangentOffset + 2])
+    );
+}
```

Comment:
Correct, since these are only two utility functions.
They both rely on different Srgs with same naming but somewhat different structure (used by different passes) and so, unless I want to use as macros, this creates a very complex inclusion scheme from two parallel source branches - it is prone to errors and I would like to avoid that at the price of these three methods..